sdk/src/main/java/com/mapbox/maps/renderer/egl/EGLCore.kt

```diff
@@ -194,40 +191,37 @@ internal class EGLCore(
     return EGL10.EGL_SUCCESS
   }
 
-  @MainThread
+  @WorkerThread
   internal fun addRendererStateListener(listener: RendererSetupErrorListener) {
-    rendererErrorLock.withLock {
-      rendererSetupErrorListenerSet.add(listener)
-      if (accumulatedRendererErrorList.isNotEmpty()) {
-        accumulatedRendererErrorList.forEach { error ->
+    rendererSetupErrorListenerSet.add(listener)
+    if (accumulatedRendererErrorList.isNotEmpty()) {
+      val immutableCopy = LinkedList(accumulatedRendererErrorList)
+      mainHandler.post {
+        immutableCopy.forEach { error ->
           listener.onError(error)
         }
-        accumulatedRendererErrorList.clear()
       }
     }
   }
 
-  @MainThread
+  @WorkerThread
   internal fun removeRendererStateListener(listener: RendererSetupErrorListener) {
-    rendererErrorLock.withLock {
-      rendererSetupErrorListenerSet.remove(listener)
-    }
+    rendererSetupErrorListenerSet.remove(listener)
```

Comment:
Maybe just post to render thread here and keep semantics of adding listeners from MainThread since we notify on main also?

Reply:
we don't have render thread instance in egl core